tests/test_config_consistency.py

```diff
@@ -276,3 +276,47 @@ def test_default_service_name(self):
         assert (
             spans[0]["service"] != "service_test"
         )  # in default scenario, DD_SERVICE is set to "weblog" in the dockerfile; this is a temp fix to test that it is not the value we manually set in the specific scenario
+
+
+@scenarios.tracing_config_nondefault
+@features.tracing_configuration_consistency
+class Test_Config_IntegrationEnabled_False:
+    """ Verify behavior of integrations automatic spans """
+
+    def setup_kafka_integration_enabled_false(self):
+        self.r = weblog.get("/kafka/produce")
+
+    def test_kafka_integration_enabled_false(self):
+        assert self.r.status_code == 200
+
+        nonKafkaSpans = []
+        kafkaSpans = []
+        # We do not use get_spans: the span we look for is not directly the span that carry the request information
+        for data, trace in interfaces.library.get_traces(request=self.r):
+            nonKafkaSpans += [(data, span) for span in trace if span.get("name") != "kafka.produce"]
+            kafkaSpans += [(data, span) for span in trace if span.get("name") == "kafka.produce"]
```

Comment:
Nit: we can use `get_spans` we just need to set `full_trace` to `True` (it's false by default)
```suggestion
        for _,  _, span in interfaces.library.get_spans(request=self.r, full_trace=True):
            if if span.get("name") != "kafka.produce"
                 nonKafkaSpans.append(span)
           else:
                 kafkaSpans.append(span)
```